ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -125,4 +216,191 @@ mod tests {
+                * f64::ln(failure_prob / 2.0),
+        );
+        println!("t: {t:?}");
+        let mut samples = Vec::new();
+        // Sample number_samples values from the generate_truncated_double_geometric function
+        let distribution = TruncatedDoubleGeometric::new(s, n)
+            .expect("Truncated Double Geometric not constructed properly");
+        for _ in 0..number_samples {
+            let sample = distribution.sample(&mut rng);
+            samples.push(sample);
+        }
+        // Compute the sample mean
+        let sample_mean = f64::from(samples.iter().sum::<u32>()) / f64::from(number_samples);
+        // println!("sample_mean: {:?}", sample_mean);
+        // Check that the sample mean is within some distance of the expected value
+        let expected_mean = f64::from(n);
+        // println!("expected_mean: {:?}", expected_mean);
+        (sample_mean >= expected_mean - t) && (sample_mean <= expected_mean + t)
+    }
+    #[test]
+    fn test_truncated_double_geometric_sample_dist() {
+        let mut rng = rand::thread_rng();
+        let epsilon = 1.0;
+        let s = 1.0 / epsilon;
+        let n = 25;
+        let distribution = TruncatedDoubleGeometric::new(s, n)
+            .expect("failed to construct TruncatedDoubleGeometric");
+        let num_samples = 100_000;
+        let mut samples = Vec::new();
+        // Sample 1000 values from the generate_truncated_double_geometric function
+        for _ in 0..num_samples {
+            let sample = distribution.sample(&mut rng);
+            assert!(sample <= 2 * n);
+            samples.push(sample);
+        }
+        // Compute the observed probability for each value in the range [0, 2*n)
+        let mut histogram = HashMap::new();
+        for value in samples {
+            *histogram.entry(value).or_insert(0) += 1;
+        }
+        let mut sorted_keys: Vec<u32> = histogram.keys().copied().collect();
+        sorted_keys.sort_unstable();
+        // Compute the expected probability for each value in the range [0, 2*n]
+        #[allow(clippy::cast_precision_loss)]
+        let normalizing_factor = (1.0 - E.powf(-epsilon))
+            / (1.0 + E.powf(-epsilon) - 2.0 * E.powf(-epsilon * (f64::from(n + 1)))); // 'A' in paper
+
+        for x in 0..=(2 * n) {
+            // Compare the observed and expected probabilities for each value in the range [0, 2*n]
+            let observed_probability = histogram
+                .get(&x)
+                .map_or(0.0, |count| f64::from(*count) / f64::from(num_samples));
+            // let expected_probability =
+            // normalizing_factor * E.powf(-epsilon * ((n - x) as i32 ).abs() as f64);
+            let expected_probability = normalizing_factor
+                * (if x <= n {
+                    E.powf(-epsilon * f64::from(n - x))
+                } else {
+                    E.powf(-epsilon * f64::from(x - n))
+                });
```

Comment:
```suggestion
                * (E.powf(-epsilon) + E.powf(abs(f64::from(n) - f64::from(x))));
```

If you upconvert the raw values, you can just use abs, just like in the comment.  And delete the comment.

Reply:
ah, that nice.  had various type conversion troubles before which led to the very manual way of doing it.